### PR TITLE
Add encoding/decoding of 'ATxAux ByteString'

### PR DESCRIPTION
**Pull request by erikd**
With a round trip test.

### CODE REVIEW

File: cardano-ledger/src/Cardano/Chain/UTxO/TxAux.hs [hunk 1]
```diff
@@ -94,8 +94,23 @@ instance FromCBOR TxAux where
...
     Annotated (tx, witness) byteSpan <- annotatedDecoder $ do
-      enforceSize "TxAux" 2
+      enforceSize "ATxAux" 2
       tx      <- fromCBORAnnotated
       witness <- fromCBORAnnotated
       pure (tx, witness)
     pure $ ATxAux tx witness byteSpan
+
+instance FromCBOR (ATxAux ByteString) where
+  fromCBOR = fmap annotateTxAux fromCBOR
```
Comment (dcoutts) [inline cardano-ledger/src/Cardano/Chain/UTxO/TxAux.hs]: We can't use `annotateTxAux` in the general case. It reconstructs the annotation by re-serialising it, but in general we need to keep the original input.

And that's the general problem with the "A" annotated variants. Annotating with `ByteSpan` is possible in a `Decoder` but annotating with the `ByteString` is not possible in a `Decoder`, which is why it needs external machinery like `fmap (BSL.toStrict . slice bytes)` with the raw input `bytes`.

**Review by dcoutts — DISMISSED** [review]
See comment below.

We can provide functions for doing this, but they cannot be `FromCBOR` instances.

**Review by JaredCorduan — APPROVED** [review]

**Review by dcoutts — APPROVED** [review]
LGTM.

File: cardano-ledger/test/Test/Cardano/Chain/UTxO/CBOR.hs [hunk 2]
```diff
@@ -351,11 +350,21 @@ sizeEstimates
...
+
 
 --------------------------------------------------------------------------------
 -- Main test export
 --------------------------------------------------------------------------------
 
 tests :: TSGroup
 tests = concatTSGroups
-  [const $$discoverGolden, $$discoverRoundTripArg, const sizeEstimates]
+  [const $$discoverGolden, const fromCborTxAuxTest, const $$discoverRoundTrip, $$discoverRoundTripArg, const sizeEstimates]
```
Comment (intricate) [inline cardano-ledger/test/Test/Cardano/Chain/UTxO/CBOR.hs]:
```suggestion
  [const $$discoverGolden, const fromCborTxAuxTest, $$discoverRoundTripArg, const sizeEstimates]
```

I believe `const $$discoverRoundTrip` is redundant as there aren't any tests in this module that are prefixed with `roundTrip`.

Comment (erikd) [reply cardano-ledger/test/Test/Cardano/Chain/UTxO/CBOR.hs]: Another good catch!

**Review by intricate — APPROVED** [review]

**Pull request closed without merging**